jax/experimental/jax2tf/tests/primitives_test.py

```diff
@@ -266,18 +266,35 @@ def test_slice(self, harness):
   @primitive_harness.parameterized(primitive_harness.lax_dynamic_slice)
   def test_dynamic_slice(self, harness):
     # JAX.dynamic_slice rejects slice sizes too big; check this, and skip jax2tf
+    args = harness.dyn_args_maker(self.rng())
+    expect_exception = {}
     if any(li - si < 0 or li - si >= sh
            for sh, si, li in zip(harness.params["shape"],
                                  harness.params["start_indices"],
                                  harness.params["limit_indices"])):
       with self.assertRaisesRegex(TypeError, ""):
-        harness.dyn_fun(*harness.dyn_args_maker(self.rng()))
-    else:
-      # TF compiler gives an error for tf.slice(start_indices < 0)
-      if any(si < 0 for si in harness.params["start_indices"]):
-        raise unittest.SkipTest("TF gives error for negative start_indices")
-      self.ConvertAndCompare(harness.dyn_fun, *harness.dyn_args_maker(self.rng()),
-                             check_compiled=False)
+        harness.dyn_fun(*args)
+      return
+
+    # TF compiler gives an error for tf.slice(start_indices < 0)
+    if any(si < 0 for si in harness.params["start_indices"]):
+      expect_exception["compiled"] = (BaseException, "")
+
+    if any(si == -100 for si in harness.params["start_indices"]):
```

Comment:
I think this "hack" should be documented or the failure case identified more precisely.

Reply:
This has been simplified, and should be cleaner now.